### PR TITLE
Remove DFE and replace with Antipatterns

### PR DESCRIPTION
**Pull request by j-baker**
As discussed.

Manual testing performed:

./gradlew publishToMavenLocal

in Salt, I set the baseline version to the artifact produced.

Validated that I had failures caused by antipatterns.

### CODE REVIEW

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy [hunk 1]
```diff
@@ -30,6 +30,7 @@ class BaselineFindBugs extends AbstractBaselinePlugin {
 
     private static final String DEFAULT_FINDBUGS_VERSION = '3.0.1'
     private static final String DEFAULT_EFFORT = 'max'
+    private static final String ANTIPATTERNS_ARTIFACT = 'com.palantir.antipatterns:antipatterns:1.0-beta.4'
```
Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: can users override the version of this? i guess you could do something like

```
configurations.findbugsPlugins.resolutionStrategy {
  force 'com.palantir.antipatterns:antipatterns:1.0'
}
```

, right? I think that's good enough.

File: gradle-baseline-java-config/resources/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -351,7 +351,6 @@
             <property name="target" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
         </module>
         <module name="CyclomaticComplexity"/> <!-- Java Coding Guidelines: Reduce Cyclomatic Complexity -->
-        <module name="DesignForExtension"/> <!-- Java Coding Guidelines: Design for extension -->
```
Comment (uschi2000) [inline gradle-baseline-java-config/resources/checkstyle/checkstyle.xml:354]: can you split this into two separate commits?

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy (hunk 1 above)
Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: need to add some docs to README.md

File: gradle-baseline-java-config/resources/checkstyle/checkstyle.xml (hunk 2 above)
Comment (jmcampanini) [inline gradle-baseline-java-config/resources/checkstyle/checkstyle.xml:354]: +1

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy (hunk 1 above)
Comment (jmcampanini) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: would def like a way for this version to be upgraded without a force resolution strategy. can we add a configuration similar to how findbugs/checkstyle works:

```
dependencies {
    antipatterns 'com.palantir.antipatterns:antipatterns:1.0'

    // or for all findbug plugins
    findbugPlugins 'com.palantir.antipatterns:antipatterns:1.0'
}
```

**Comment by uschi2000** [comment]
ping

**Comment by uschi2000** [comment]
are you going to get to this, James?

**Comment by uschi2000** [comment]
@j-baker

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy (hunk 1 above)
Comment (j-baker) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: Can't you just override your antipatterns version by just setting

```
findbugsPlugins 'com.palantir.antipatterns:1.1.0'
```

or whatever?

Won't that override it?

Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: you may need to force that version (see my comment above)

Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: I'm happy with the way it is now though

File: gradle-baseline-java-config/resources/checkstyle/checkstyle.xml (hunk 2 above)
Comment (j-baker) [inline gradle-baseline-java-config/resources/checkstyle/checkstyle.xml:354]: done

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy (hunk 1 above)
Comment (j-baker) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: I just spent a while fiddling with versions (publishToMavenLocalYay), and I can verify that overriding in the way I document works, although it obviously only works for forward versions.

Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: ok, cool. last point: can I selectively disable the antipatterns plugin if I want to?

Comment (j-baker) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFindBugs.groovy:33]: yeah - just add exclusions for those checks (in the standard findbugs-exclusions.xml way)

**Comment by uschi2000** [comment]
Thanks, James!
